zap/src/test/java/org/zaproxy/zap/authentication/AuthenticationMethodPollUrlUnitTest.java

```diff
@@ -0,0 +1,219 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.authentication.AuthenticationMethod.AuthCheckingStrategy;
+import org.zaproxy.zap.authentication.AuthenticationMethod.AuthPollFrequencyUnits;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+import org.zaproxy.zap.testutils.TestUtils;
+
+@ExtendWith(MockitoExtension.class)
+public class AuthenticationMethodPollUrlUnitTest extends TestUtils {
+
+    private static final String LOGGED_IN_INDICATOR = "logged in";
+    private static final String LOGGED_IN_BODY =
+            "Lorem ipsum dolor sit amet, consectetur adipiscing elit. "
+                    + "Pellentesque auctor nulla id turpis placerat vulputate."
+                    + LOGGED_IN_INDICATOR
+                    + " Proin tempor bibendum eros rutrum. ";
+
+    private HttpMessage loginMessage;
+    private AuthenticationMethod method;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        loginMessage = new HttpMessage();
+        HttpRequestHeader header = new HttpRequestHeader();
+        header.setURI(new URI("http://www.example.com", true));
+        loginMessage.setRequestHeader(header);
+        method = Mockito.mock(AuthenticationMethod.class, Mockito.CALLS_REAL_METHODS);
+        method.setAuthCheckingStrategy(AuthCheckingStrategy.EACH_RESP);
+
+        this.startServer();
+    }
+
+    @AfterEach
+    public void shutDownServer() throws Exception {
+        stopServer();
+    }
+
+    @Test
+    public void shouldPollOnFirstRequest() throws NullPointerException, IOException {
+        // Given
+        String test = "/shouldPollOnFirstRequest/test";
+        String pollUrl = "/shouldPollOnFirstRequest/pollUrl";
+        final List<String> orderedReqs = new ArrayList<>();
+
+        this.nano.addHandler(
+                new NanoServerHandler(pollUrl) {
+                    @Override
+                    protected Response serve(IHTTPSession session) {
+                        orderedReqs.add(session.getUri());
+                        return newFixedLengthResponse(LOGGED_IN_BODY);
+                    }
+                });
+        HttpMessage testMsg = this.getHttpMessage(test);
+        HttpMessage pollMsg = this.getHttpMessage(pollUrl);
+
+        method.setAuthCheckingStrategy(AuthCheckingStrategy.POLL_URL);
+        method.setPollUrl(pollMsg.getRequestHeader().getURI().toString());
+        method.setPollFrequencyUnits(AuthPollFrequencyUnits.REQUESTS);
+        method.setPollFrequency(5);
+        method.setLoggedInIndicatorPattern(LOGGED_IN_INDICATOR);
+
+        // When/Then
+        assertThat(method.isAuthenticated(testMsg, null), is(true));
+        assertThat(orderedReqs.size(), is(1));
+        assertThat(orderedReqs.get(0), is(pollUrl));
+    }
+
+    @Test
+    public void shouldPollEveryFiveRequests() throws NullPointerException, IOException {
```

Comment:
`EveryFive` → `OnSpecifiedNumberOf` (or something), that's using five for the tests is not the important.